localstack/services/kinesis/kinesis_starter.py

```diff
@@ -1,214 +1,80 @@
-            initialize_streams_param,
-            kinesis_mock_bin,
-        )
-
-    return _run_proxy_and_command(
-        cmd=cmd,
-        port=port,
-        backend_port=backend_port,
-        update_listener=update_listener,
-        asynchronous=asynchronous,
-    )
+            raise ValueError('Unsupported Kinesis provider "%s"' % config.KINESIS_PROVIDER)
+    _start_kinesis_helper(_server, port=port, update_listener=update_listener)
+    return _server
 
 
-def start_kinesalite(port=None, asynchronous=False, update_listener=None):
-    # install and apply patches
-    install.install_kinesalite()
-    apply_patches_kinesalite()
-    # start up process
-    backend_port = get_free_tcp_port()
-    global PORT_KINESIS_BACKEND
-    PORT_KINESIS_BACKEND = backend_port
-    latency = config.KINESIS_LATENCY
-    kinesis_data_dir_param = ""
-    if config.dirs.data:
-        kinesis_data_dir = "%s/kinesis" % config.dirs.data
-        mkdir(kinesis_data_dir)
-        kinesis_data_dir_param = "--path %s" % kinesis_data_dir
-    cmd = (
-        "%s/node_modules/kinesalite/cli.js --shardLimit %s --port %s"
-        " --createStreamMs %s --deleteStreamMs %s --updateStreamMs %s %s"
-    ) % (
-        MODULE_MAIN_PATH,
-        config.KINESIS_SHARD_LIMIT,
-        backend_port,
-        latency,
-        latency,
-        latency,
-        kinesis_data_dir_param,
-    )
-
-    return _run_proxy_and_command(
-        cmd=cmd,
-        port=port,
-        backend_port=backend_port,
+def _start_kinesis_helper(server: Server, port=None, update_listener=None) -> Server:
+    server.start()
+    log_startup_message("Kinesis")
+    port = port or config.PORT_KINESIS
+    start_proxy_for_service(
+        "kinesis",
+        port,
+        backend_port=_server.port,
         update_listener=update_listener,
-        asynchronous=asynchronous,
     )
 
 
 def check_kinesis(expect_shutdown=False, print_error=False):
-    if expect_shutdown is False and kinesis_stopped.is_set():
-        raise AssertionError("kinesis backend has stopped")
-
     out = None
+    if not expect_shutdown:
+        assert _server
+
     try:
-        # check Kinesis
-        wait_for_port_open(PORT_KINESIS_BACKEND, http_path="/", expect_success=False, sleep_time=1)
-        endpoint_url = f"http://127.0.0.1:{PORT_KINESIS_BACKEND}"
+        _server.wait_is_up()
         out = aws_stack.connect_to_service(
-            service_name="kinesis", endpoint_url=endpoint_url
+            service_name="kinesis", endpoint_url=_server.url
         ).list_streams()
     except Exception:
         if print_error:
             LOG.exception("Kinesis health check failed")
-
     if expect_shutdown:
-        assert out is None or kinesis_stopped.is_set()
+        assert out is None
     else:
-        assert not kinesis_stopped.is_set()
-        assert out and isinstance(out.get("StreamNames"), list)
-
-
-def kinesis_running():
-    return PROCESS_THREAD is not None
```

Comment:
Unfortunately, this method is used in one of our extensions, so this method should stay.
Should be fairly easy though, since the `_server` is only set if the server is running.

Reply:
Thanks for teaching me that the name of that format is `reST`! We used that at my previous employer and I like it better than what I put up here, which is just based on the first example that I pulled up :)